pkg/registry/common/connect/nse_server_test.go

```diff
@@ -67,31 +61,34 @@ func TestConnect_NewNetworkServiceEndpointRegistryServer(t *testing.T) {
 
 	s := connect.NewNetworkServiceEndpointRegistryServer(ctx, func(_ context.Context, cc grpc.ClientConnInterface) registry.NetworkServiceEndpointRegistryClient {
 		return registry.NewNetworkServiceEndpointRegistryClient(cc)
-	}, connect.WithExpirationDuration(time.Millisecond*100), connect.WithClientDialOptions(grpc.WithInsecure()))
+	}, grpc.WithInsecure())
+
+	_, err := s.Register(clienturlctx.WithClientURL(context.Background(), url1), &registry.NetworkServiceEndpoint{Name: "nse-1"})
+	require.NoError(t, err)
+
+	_, err = s.Register(clienturlctx.WithClientURL(context.Background(), url2), &registry.NetworkServiceEndpoint{Name: "nse-1-1"})
+	require.NoError(t, err)
 
-	_, err := s.Register(clienturlctx.WithClientURL(context.Background(), url1), &registry.NetworkServiceEndpoint{Name: "ns-1"})
-	require.Nil(t, err)
-	_, err = s.Register(clienturlctx.WithClientURL(context.Background(), url2), &registry.NetworkServiceEndpoint{Name: "ns-1-1"})
-	require.Nil(t, err)
 	ch := make(chan *registry.NetworkServiceEndpoint, 1)
 	findSrv := streamchannel.NewNetworkServiceEndpointFindServer(clienturlctx.WithClientURL(context.Background(), url1), ch)
 	err = s.Find(&registry.NetworkServiceEndpointQuery{NetworkServiceEndpoint: &registry.NetworkServiceEndpoint{
-		Name: "ns-1",
+		Name: "nse-1",
 	}}, findSrv)
-	require.Nil(t, err)
-	require.Equal(t, (<-ch).Name, "ns-1")
+	require.NoError(t, err)
+	require.Equal(t, (<-ch).Name, "nse-1")
+
 	findSrv = streamchannel.NewNetworkServiceEndpointFindServer(clienturlctx.WithClientURL(context.Background(), url2), ch)
 	err = s.Find(&registry.NetworkServiceEndpointQuery{NetworkServiceEndpoint: &registry.NetworkServiceEndpoint{
-		Name: "ns-1",
+		Name: "nse-1",
 	}}, findSrv)
-	require.Nil(t, err)
-	require.Equal(t, (<-ch).Name, "ns-1-1")
+	require.NoError(t, err)
+	require.Equal(t, (<-ch).Name, "nse-1-1")
 
 	closeServer1()
 	closeServer2()
 
-	require.Eventually(t, func() bool {
-		runtime.GC()
-		return goleak.Find() != nil
-	}, time.Second, time.Microsecond*100)
+	var i int
+	for err, i = goleak.Find(), 0; err != nil && i < 3; err, i = goleak.Find(), i+1 {
+	}
```

Comment:
Please use Eventually

Reply:
Here is a problem with `Eventually` and `goleak.Find()`. `goleak.Find()` takes a lot of time, so such thing:
```go
require.Eventually(t, func() bool {
	return goleak.Find(goleak.IgnoreTopFunction("github.com/stretchr/testify/assert.Eventually")) == nil
}, time.Second, 100*time.Millisecond)
```
will be executed only 2-3 times on Ubuntu, MacOS on [~100ms, ~400ms, ~900ms] and only once on Windows on [~100ms] (it is not enough, test always fails on Windows). So we either need to use it with some relatively huge timeout (2-3s) or just don't use it.

What do you think?

Reply:
And so even with greatest timeout we doesn't guarantee anything for the VERY slow environment :)